portal/portal-impl/impl/src/java/org/sakaiproject/portal/entityprovider/PortalEntityProvider.java

```diff
@@ -123,6 +127,126 @@ public PortalNotifications handleNotify(EntityView view) {
 		return noti;
 	}
 
+	@EntityCustomAction(action = "socialAlerts", viewKey = EntityView.VIEW_LIST)
+	public ActionReturn getSocialAlerts(EntityView view) {
+
+		String currentUserId = getCheckedCurrentUser();
+
+		List<BullhornAlert> alerts = portalService.getSocialAlerts(currentUserId);
+
+		ResourceLoader rl = new ResourceLoader("bullhorns");
+
+		if (alerts.size() > 0) {
+			Map<String, Object> data = new HashMap();
+			data.put("alerts", alerts);
+			data.put("i18n", rl);
+
+			return new ActionReturn(data);
+		} else {
+			Map<String, String> i18n = new HashMap();
+			i18n.put("noAlerts", rl.getString("noAlerts"));
+
+			Map<String, Object> data = new HashMap();
+			data.put("message", "NO_ALERTS");
+			data.put("i18n", i18n);
+
+			return new ActionReturn(data);
+		}
+	}
+
+	@EntityCustomAction(action = "clearBullhornAlert", viewKey = EntityView.VIEW_LIST)
+	public boolean clearBullhornAlert(Map<String, Object> params) {
+
+		String currentUserId = getCheckedCurrentUser();
+
+		try {
+			long alertId = Long.parseLong((String) params.get("id"));
+			return portalService.clearBullhornAlert(currentUserId, alertId);
+		} catch (Exception e) {
+			log.error("Failed to clear social alert", e);
+		}
+
+		return false;
+	}
+
+	@EntityCustomAction(action = "clearAllSocialAlerts", viewKey = EntityView.VIEW_LIST)
+	public boolean clearAllSocialAlerts(Map<String, Object> params) {
+
+		String currentUserId = getCheckedCurrentUser();
+
+		try {
+			return portalService.clearAllSocialAlerts(currentUserId);
+		} catch (Exception e) {
+			log.error("Failed to clear all social alerts", e);
+		}
+
+		return false;
+	}
+
+	@EntityCustomAction(action = "academicAlerts", viewKey = EntityView.VIEW_LIST)
+	public ActionReturn getAcademicAlerts(EntityView view) {
+
+		String currentUserId = getCheckedCurrentUser();
```

Comment:
Seems like the code in this method and `getSocialAlerts()` are identical; opportunity to extract and reduce code duplication.